### PR TITLE
Add ordinal to merge schema after merge

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
Still need to add test cases here

### CODE REVIEW

File: go/libraries/doltcore/merge/merge_schema.go [hunk 1]
```diff
@@ -118,6 +118,12 @@ func SchemaMerge(ourSch, theirSch, ancSch schema.Schema, tblName string) (sch sc
 	if err != nil {
 		return nil, sc, err
 	}
+
+	err = sch.SetPkOrdinals(ourSch.GetPkOrdinals())
```
Comment (max-hoffman) [inline go/libraries/doltcore/merge/merge_schema.go:122]: what if the three schemas have different pk ordinals? how to tell which one to keep?

Comment (VinaiRachakonda) [reply go/libraries/doltcore/merge/merge_schema.go:124]: I believe we do not support merge between tables with competing ordinals.

File: go/libraries/doltcore/sqle/enginetest/dolt_transaction_queries.go [hunk 2]
```diff
@@ -738,4 +738,57 @@ var DoltTransactionTests = []enginetest.TransactionTest{
...
+			{
+				Query:    "/* client a */ select * from test order by x",
+				Expected: []sql.Row{{1, 1, 1}, {2, 3, 2}, {2, 5, 2}},
+			},
+			{
+				Query:    "/* client b */ select * from test order by x",
+				Expected: []sql.Row{{1, 1, 1}, {2, 3, 2}, {2, 5, 2}},
+			},
+			{
+				Query:       "/* client b */ insert into test values (4,3,2)",
```
Comment (max-hoffman) [inline go/libraries/doltcore/sqle/enginetest/dolt_transaction_queries.go:789]: doesn't strike me as the most direct way to test the change, but if it mimics the original customer issue then it make sense

File: go/libraries/doltcore/merge/merge_schema.go [hunk 3]
```diff
@@ -118,6 +118,14 @@ func SchemaMerge(ourSch, theirSch, ancSch schema.Schema, tblName string) (sch sc
 	if err != nil {
 		return nil, sc, err
 	}
+
+	// TODO: We should enforce the existing primary key set pk ordinals as out of order primary
+	// key ordinal merge is not supported yet.
```
Comment (max-hoffman) [inline go/libraries/doltcore/merge/merge_schema.go:123]: "merge conflict should have blocked any ordinal change"?

**Review by max-hoffman — APPROVED** [review]
LGTM, a simpler unit test would have been helpful, the script test seems sufficient

**Review by max-hoffman — APPROVED** [review]
LGTM, a simpler unit test would have been helpful, the script test seems sufficient

File: go/libraries/doltcore/sqle/enginetest/dolt_transaction_queries.go (hunk 2 above)
Comment (VinaiRachakonda) [reply go/libraries/doltcore/sqle/enginetest/dolt_transaction_queries.go:789]: Yeah without the change, this becomes an incorrect primary key issue and the ordering of the select changes. This indicates that primary key set has changed.

**Pull request merged**